src/Microsoft.IdentityModel.Tokens/EventBasedLRUCache.cs

```diff
@@ -309,6 +325,8 @@ public bool TryRemove(TKey key, out TValue value)
                 return true;
             }
 
+            // start the event queue task if it is not running
+            StartEventQueueTaskIfNotRunning();
```

Comment:
Adding this to the Get() and Remove() APIs was necessary to avoid a build up of Actions  in the _eventQueue that were never processed (was causing memory issues).

Reply:
Good catch on starting the task anywhere new events are added. But for line 329 in TryRemove() do we need to start the task?

Reply:
Fixed!

Thanks for this suggestion Danny! I think this will really help future proof the code and prevent developer mistakes. :)